src/core/Database.cpp

```diff
@@ -292,6 +292,21 @@ bool Database::writeDatabase(QIODevice* device, QString* error)
     return true;
 }
 
+bool Database::extract(QByteArray& xmlOutput, QString* error)
+{
+    KeePass2Writer writer;
+    writer.extractDatabase(this, xmlOutput);
+
+    if (writer.hasError()) {
+        // the writer failed
```

Comment:
Obviously. ;-)

Reply:
@phoerious not a fan of those comments either, I just carried it over from the error handling code of `saveDatabase`. Want it gone :hocho: ??

Reply:
It's silly, but no big deal. Remove it if you want. I don't think it documents anything that's not totally obvious already.